lib/rules/no-extend-native.js

```diff
@@ -9,9 +9,16 @@
 // Requirements
 //------------------------------------------------------------------------------
 
+const astUtils = require("../ast-utils");
 const globals = require("globals");
 
 //------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const propertyDefinitionMethods = new Set(["defineProperty", "defineProperties"]);
```

Comment:
`Object.assign` isn't currently covered by this rule. Okay if I address that in a separate PR?